Issues with Backport.System.Threading.Lock

Somethings going wrong with the package `Backport.System.Threading.Lock`. Also since it's not using file-scoped namespaces, it throws of our formatting rules.

```cs
 warning: Backport.System.Threading.Lock.SourceGenerator/Backport.System.Threading.Lock.SourceGenerator.Generator/Backport.System.Threading.Lock.SourceGenerator.Lock.cs(105,33): warning CS0436: The type 'Lock' in 'Backport.System.Threading.Lock.SourceGenerator/Backport.System.Threading.Lock.SourceGenerator.Generator/Backport.System.Threading.Lock.SourceGenerator.Lock.cs' conflicts with the imported type 'Lock' in 'DisCatSharp, Version=10.6.6.0, Culture=neutral, PublicKeyToken=null'. Using the type defined in 'Backport.System.Threading.Lock.SourceGenerator/Backport.System.Threading.Lock.SourceGenerator.Generator/Backport.System.Threading.Lock.SourceGenerator.Lock.cs'.
warning: Backport.System.Threading.Lock.SourceGenerator/Backport.System.Threading.Lock.SourceGenerator.Generator/Backport.System.Threading.Lock.SourceGenerator.PreNet5Lock.cs(141,33): warning CS0436: The type 'Lock' in 'Backport.System.Threading.Lock.SourceGenerator/Backport.System.Threading.Lock.SourceGenerator.Generator/Backport.System.Threading.Lock.SourceGenerator.PreNet5Lock.cs' conflicts with the imported type 'Lock' in 'DisCatSharp, Version=10.6.6.0, Culture=neutral, PublicKeyToken=null'. Using the type defined in 'Backport.System.Threading.Lock.SourceGenerator/Backport.System.Threading.Lock.SourceGenerator.Generator/Backport.System.Threading.Lock.SourceGenerator.PreNet5Lock.cs'.
warning: Backport.System.Threading.Lock.SourceGenerator/Backport.System.Threading.Lock.SourceGenerator.Generator/Backport.System.Threading.Lock.SourceGenerator.LockFactory.cs(46,23): warning CS0436: The type 'Lock' in 'Backport.System.Threading.Lock.SourceGenerator/Backport.System.Threading.Lock.SourceGenerator.Generator/Backport.System.Threading.Lock.SourceGenerator.PreNet5Lock.cs' conflicts with the imported type 'Lock' in 'DisCatSharp, Version=10.6.6.0, Culture=neutral, PublicKeyToken=null'. Using the type defined in 'Backport.System.Threading.Lock.SourceGenerator/Backport.System.Threading.Lock.SourceGenerator.Generator/Backport.System.Threading.Lock.SourceGenerator.PreNet5Lock.cs'.
```
 
CC @MarkCiliaVincenti

Will investigate.

On Fri, 6 Dec 2024, 23:06 Lala Sabathil, ***@***.***> wrote:

> Somethings going wrong with the package Backport.System.Threading.Lock.
> Also since it's not using file-scoped namespaces, it throws of our
> formatting rules.
>
>  warning: Backport.System.Threading.Lock.SourceGenerator/Backport.System.Threading.Lock.SourceGenerator.Generator/Backport.System.Threading.Lock.SourceGenerator.Lock.cs(105,33): warning CS0436: The type 'Lock' in 'Backport.System.Threading.Lock.SourceGenerator/Backport.System.Threading.Lock.SourceGenerator.Generator/Backport.System.Threading.Lock.SourceGenerator.Lock.cs' conflicts with the imported type 'Lock' in 'DisCatSharp, Version=10.6.6.0, Culture=neutral, PublicKeyToken=null'. Using the type defined in 'Backport.System.Threading.Lock.SourceGenerator/Backport.System.Threading.Lock.SourceGenerator.Generator/Backport.System.Threading.Lock.SourceGenerator.Lock.cs'.warning: Backport.System.Threading.Lock.SourceGenerator/Backport.System.Threading.Lock.SourceGenerator.Generator/Backport.System.Threading.Lock.SourceGenerator.PreNet5Lock.cs(141,33): warning CS0436: The type 'Lock' in 'Backport.System.Threading.Lock.SourceGenerator/Backport.System.Threading.Lock.SourceGenerator.Generator/Backport.System.Threading.Lock.SourceGenerator.PreNet5Lock.cs' conflicts with the imported type 'Lock' in 'DisCatSharp, Version=10.6.6.0, Culture=neutral, PublicKeyToken=null'. Using the type defined in 'Backport.System.Threading.Lock.SourceGenerator/Backport.System.Threading.Lock.SourceGenerator.Generator/Backport.System.Threading.Lock.SourceGenerator.PreNet5Lock.cs'.warning: Backport.System.Threading.Lock.SourceGenerator/Backport.System.Threading.Lock.SourceGenerator.Generator/Backport.System.Threading.Lock.SourceGenerator.LockFactory.cs(46,23): warning CS0436: The type 'Lock' in 'Backport.System.Threading.Lock.SourceGenerator/Backport.System.Threading.Lock.SourceGenerator.Generator/Backport.System.Threading.Lock.SourceGenerator.PreNet5Lock.cs' conflicts with the imported type 'Lock' in 'DisCatSharp, Version=10.6.6.0, Culture=neutral, PublicKeyToken=null'. Using the type defined in 'Backport.System.Threading.Lock.SourceGenerator/Backport.System.Threading.Lock.SourceGenerator.Generator/Backport.System.Threading.Lock.SourceGenerator.PreNet5Lock.cs'.
>
> CC @MarkCiliaVincenti <https://github.com/MarkCiliaVincenti>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Aiko-IT-Systems/DisCatSharp/issues/595>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AF7U7YCNQHYJLGTAOPAHI2L2EINXTAVCNFSM6AAAAABTFP2LGKVHI2DSMVQWIX3LMV43ASLTON2WKOZSG4ZDIMBQHA3DSNY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

--- a/DisCatSharp/DisCatSharp.csproj
+++ b/DisCatSharp/DisCatSharp.csproj
@@ -26,7 +26,7 @@
 	</PropertyGroup>
 
 	<ItemGroup Condition="!$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net9.0'))">
-		<PackageReference Include="Backport.System.Threading.Lock" Version="3.0.3" />
+		<PackageReference Include="Backport.System.Threading.Lock" Version="3.1.0" />
 	</ItemGroup>
 
 	<ItemGroup>

